### ISSUE TITLE
Multiple coverage reports output for multitask

### ISSUE THREAD

**Issue by deckar01** [maenu/grunt-template-jasmine-istanbul#58]
I am taking advantage of the multitask functionality of grunt that `grunt-contrib-jasmine` supports to separate each spec file into its own task so that each set of vendor dependencies is isolated.

This template treats each task as a separate coverage report, which is a reasonable default behavior, but it would be useful to me to be able to defer the report until after all the tasks have finished and allow the report to accrue into a more traditional coverage report.

It would be nice to have a config option like `multitaskReport`.

**Comment by maenu** [comment]
I think this is a duplicate of #57?
I'd propose to use a solution with [grunt-istanbul-combine](https://www.npmjs.com/package/grunt-istanbul-combine), as the main purpose of this plugin is to glue jasmine and istanbul together to collect coverage data. The whole reporting feature of this plugin is just transparently reflected onto istanbul itself as it was trivial to do, I don't see it as a core functionality. Yet I see your point that the only output we are interested in, is the actual report, not the unreadable coverage data itself. Nevertheless, I don't see how a `multitaskReport ` option could be inserted in this plugin: Before any task is run, the old coverage data has to be deleted from someone. After all tasks have run, the report should be generated. Those are problems, a single task can not take responsibility for, as they would need synchronization. How would that work in a concurrent environment otherwise? The solution would be to create another plugin that works like this: delete obsolete coverage data, run multi-task, collect coverage data, generate report.

**Comment by deckar01** [comment]
@maenu This is basically a duplicate of #57, I should have looked at the closed issues.

Thank you for your insight. That is a reasonable solution.

For future travelers, I am programatically setting the `templateOptions` on each child task with a namespaced coverage value and setting the report value to an empty array to skip the individual reports.

```js
Object.keys(config).forEach(function(key) {
  if(key == 'options') return;
  config[key].options.templateOptions = {
    coverage: 'bin/coverage/' + key + '.coverage.json',
    report: []
  };
});
```

Then in my `Gruntfile.js` I run the `grunt_combine` task after `jasmine` to output the combined report.

```js
grunt.initConfig({
  istanbul_combine: {
    options: {
      dir: 'bin/coverage/report',
      pattern: 'bin/coverage/*.coverage.json',
      print: 'both',
      reporters: { html: {} }
    }
  }
});

grunt.loadNpmTasks('grunt-istanbul-combine');
grunt.registerTask('test', ['jasmine', 'istanbul_combine']);
```

Thanks again for your help @maenu!

**Comment by maenu** [comment]
That's already quite a generalized solution. You could certainly wrap that in a plugin so others can use it without the hassle.

**Comment by deckar01** [comment]
I will look into that. I already created one tiny jasmine helper today :)

https://github.com/deckar01/jasmine-matcher-errors
